### ISSUE TITLE
Xcode 7/Swift 2.0 Failed to generate documentation

### ISSUE THREAD

**Issue by brandoncgray** [realm/jazzy#319]
Xcode 6 worked just fine but Xcode 7 doesn't work. Here is part of my bash file:

"`dirname "$0"`"/jazzy-master/bin/jazzy \
--output "${outputName}" \
--clean \
--author_url "${companyURL}" \
--module "XXXXXXX" \
--source-directory "`dirname "$0"`" \
--copyright "${copyright}" \
--swift-version "2.0" \
--min-acl "internal" \
-x -scheme,"XXXXXXX Prod" \
--skip-undocumented  

This is the results in terminal, I have a bash file that makes sure that everything is installed for others to easily run:

brew Found 

Already up-to-date.
Warning: sourcekitten-0.5.0 already installed

rvm Found

RVM reloaded!
Already installed ruby-2.0.0-p643.
To reinstall use:

    rvm reinstall ruby-2.0.0-p643

Using /Users/XXXXXXX/.rvm/gems/ruby-2.0.0-p643
ruby 2.0.0p643 (2015-02-25 revision 49749) [x86_64-darwin14.1.0]

bundler Found

Resolving dependencies...
Using rake 10.3.2
Using i18n 0.7.0
Using json 1.8.3
Using minitest 5.8.0
Using thread_safe 0.3.5
Using tzinfo 1.2.2
Using activesupport 4.2.4
Using addressable 2.3.6
Using ast 2.0.0
Using slop 3.6.0
Using parser 2.2.0.pre.5
Using astrolabe 1.3.0
Using bacon 1.2.0
Using bundler 1.10.6
Using claide 0.9.1
Using colored 1.2
Using diffy 3.0.7
Using clintegracon 0.5.3
Using fuzzy_match 2.0.4
Using nap 0.8.0
Using cocoapods-core 0.38.2
Using cocoapods-downloader 0.9.3
Using cocoapods-plugins 0.4.2
Using cocoapods-stats 0.5.3
Using netrc 0.7.8
Using cocoapods-trunk 0.6.4
Using cocoapods-try 0.4.5
Using escape 0.0.4
Using molinillo 0.3.1
Using xcodeproj 0.26.3
Using cocoapods 0.38.2
Using safe_yaml 1.0.4
Using crack 0.4.2
Using mustache 0.99.8
Using open4 1.3.4
Using redcarpet 3.3.2
Using rouge 1.9.0
Using sass 3.4.18
Using sqlite3 1.3.10
Using liferaft 0.0.4
Using xcinvoke 0.2.0
Using jazzy 0.3.1 from source at .
Using metaclass 0.0.4
Using mocha 1.1.0
Using mocha-on-bacon 0.2.2
Using powerpack 0.0.9
Using prettybacon 0.0.2
Using rainbow 2.0.0
Using ruby-progressbar 1.6.0
Using rubocop 0.26.1
Using webmock 1.20.0
Bundle complete! 11 Gemfile dependencies, 51 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
Running xcodebuild

Could not parse compiler arguments from `xcodebuild` output.
Build settings from command line:
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO

=== CLEAN TARGET XXXXXXX OF PROJECT XXXXXXX WITH CONFIGURATION Release ===

Check dependencies

** CLEAN SUCCEEDED **

=== BUILD TARGET XXXXXXX OF PROJECT XXXXXXX WITH CONFIGURATION Release ===

Check dependencies

Write auxiliary files
write-file /Users/XXXXXXX/Library/Developer/Xcode/DerivedData/XXXXXXX-fuvckrilkvghklagfoohielidrvk/Build/Intermediates/XXXXXXX.build/Release-iphoneos/XXXXXXX.build/XXXXXXX-project-headers.hmap
......
*Note - continues listing lots of directories
** BUILD SUCCEEDED **

Failed to generate documentation
	from /Users/XXXXXXX/Documents/Repos/XXXXXXX/jazzy-master/lib/jazzy/sourcekitten.rb:128:in `run_sourcekitten'
	from /Users/XXXXXXX/Documents/Repos/XXXXXXX/jazzy-master/lib/jazzy/doc_builder.rb:56:in `block in build'
	from /Users/XXXXXXX/Documents/Repos/XXXXXXX/jazzy-master/lib/jazzy/doc_builder.rb:54:in `chdir'
	from /Users/XXXXXXX/Documents/Repos/XXXXXXX/jazzy-master/lib/jazzy/doc_builder.rb:54:in `build'
	from /Users/XXXXXXX/Documents/Repos/XXXXXXX/jazzy-master/bin/jazzy:15:in `<main>'

Clean Up

rm: /Users/XXXXXXX/Documents/Repos/XXXXXXX/build: No such file or directory

Finished

One thing that I notice is that the build folder that I was removing when I ran Jazzy for Xcode 6/Swift 1.2 is not being created in the project folder anymore. Also I am getting sourcekitten from homebrew and I have never got the  CODE_SIGN_IDENTITY = and the CODE_SIGNING_REQUIRED = NO warnings before with Xcode 6/Swift 1.2.

**Comment by jpsim** [comment]
Could you please print the result of the following command? `xcodebuild -scheme "XXXXXXX Prod"`

**Comment by brandoncgray** [comment]
It is a massive amount of text, it wont let me paste it here... I also tried to attach it and it won't attach. Is there a specific part you are looking for?

**Comment by jpsim** [comment]
> Is there a specific part you are looking for?

The part that's making this fail... You should be able to post it in a [gist](https://gist.github.com) and paste the link here.

**Comment by brandoncgray** [comment]
https://www.dropbox.com/s/0j1khlvga5abdab/xcode_build_results.txt?dl=0

**Comment by jpsim** [comment]
Alright, it looks like the scheme name ("XXXXXXX Prod") doesn't match the module name ("XXXXXXX Prod"), causing jazzy (well, SourceKitten really) to not find which module to build docs for. Could you please try passing in `--module-name XXXXXXX` to jazzy?

**Comment by brandoncgray** [comment]
I am getting this error: invalid option: --module-name (OptionParser::InvalidOption)

**Comment by jpsim** [comment]
That's my mistake, I didn't remember the name of the parameter. It's `--module` not `--module-name`

**Comment by brandoncgray** [comment]
Ok no problem, so this is what I am using to run jazzy. I was already using the `--module` in the file you looked at. 

# Start constants
company="Some Company";
companyID="Some Company ID";
companyURL="http://www.SomeCompany.com";
outputName="~/Documents/Auto Documentation";
copyright="© 2015 All Rights Reserved.";
appName="XXXXXXX";
swiftVersion="2.0";
# End constants

"`dirname "$0"`"/jazzy-master/bin/jazzy

# Start Jazzy Document Creation
"`dirname "$0"`"/jazzy-master/bin/jazzy \
--output "${outputName}" \
--clean \
--author "Brandon Gray" \
--author_url "${companyURL}" \
--module "${appName}" \
--source-directory "`dirname "$0"`" \
--copyright "${copyright}" \
--swift-version "${swiftVersion}" \
--min-acl "internal" \
-x -scheme,"${appName} Prod" \
--module-name "${appName}" \
--skip-undocumented

**Comment by brandoncgray** [comment]
So I commented the `--swift-version "${swiftVersion}" \` line and I actually saw the build folder get created and I received this error. The app(My App) has spaces in it and for some reason "My\\ App" has double slashes instead of a single slash.

```
Failed to generate documentation
/Users/brandongray/Documents/Repos/My App/jazzy-master/lib/jazzy/executable.rb:35:in `execute_command': /Users/brandon.gray/Documents/Repos/My\ App/jazzy-master/lib/jazzy/SourceKitten/bin/sourcekitten ["doc", "--module-name", “My\\ App”] (RuntimeError)
```

**Comment by brandoncgray** [comment]
Ok So I got it working... --module "My App Name" brakes it. In other words spaces in the app name looks like the cause of the issue. So apps without a space in the name would work properly. I changed my code and this works minus the name of my app on the jazzy site.

```
# Start Jazzy Document Creation
"`dirname "$0"`"/jazzy-master/bin/jazzy \
--output "${outputName}" \
--clean \
--author "Brandon Gray" \
--author_url "${companyURL}" \
--source-directory "`dirname "$0"`" \
--copyright "${copyright}" \
--swift-version "${swiftVersion}" \
--min-acl "internal" \
--skip-undocumented 
```

**Comment by jpsim** [comment]
I believe this is actually the same issue as #259, which was addressed by #298 but has yet to be included in a release. I'll try to cut a release tonight since it's been about a month since our last one.

I'm closing this issue in the hopes that it's the one I recently fixed, but please reopen if you manage to reproduce it again with 0.3.2, which should be out shortly.

**Comment by jpsim** [comment]
0.3.2 is out: https://github.com/realm/jazzy/releases/tag/v0.3.2
